// random-chars/distribute/random-chars.js
//Default chars there
var _defaultChars = {
	lower: 'abcdefghijklmnopqrstuvwxyz',
	upper: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
	digits: '1234567890'
}

//randomChars module
var randomChars = {
	get: function(op, de) {
		var options = {
			amount: 16,
			chars: '',
			use: {
				lower: true,
				upper: true,
				digits: true
			}
		}, result = '';
		if(typeof op == 'object') {
			//amount
			if(op.amount) {
				options.amount = op.amount;
			}
			//chars
			if(op.chars) {
				options.chars = op.chars;
			}
			else {
				//useables
				if(typeof op.use == 'object') {
					options.use.lower = (typeof op.use.lower == 'boolean') ? op.use.lower : true;
					options.use.upper = (typeof op.use.upper == 'boolean') ? op.use.upper : true;
					options.use.digits = (typeof op.use.digits == 'boolean') ? op.use.digits : true;
				}

				//uppercase
				if(options.use.lower) {
					options.chars += _defaultChars.lower;
				}
				//lowercase
				if(options.use.upper) {
					options.chars += _defaultChars.upper;
				}
				//digits
				if(options.use.digits) {
					options.chars += _defaultChars.digits;
				}
			}
		}
		else {
			if(typeof op == 'number') {
				options.amount = op;
			}
			if(typeof de == 'string') {
				options.chars = de;
			}
			else options.chars = _defaultChars.lower + _defaultChars.upper + _defaultChars.digits;
		}
		//generate result
		for(var i = 0; i < options.amount; i++) {
			result += options.chars.charAt(Math.floor(Math.random() * options.chars.length));
		}
		return result;
	}
}

//CommonJS/Node.js
if(typeof module == 'object' && typeof module.exports == 'object') {
	module.exports = randomChars;
}

//AMD
else if(typeof define == 'function' && define.amd) {
	define([], randomChars);
}

//Browsers
else {
	window.randomChars = randomChars;
}
